Add option to disable using hovered directory

nralbrecht opened the pull request:
Coming from Ranger, I find it very unintuitive that the hovered file is used to create a new bookmark.

I added a new option called `bookmark_file_pick_mode`. With this one can configure to always use the currently open directory when set to `parent`. When set to `hover` the current default behavior ist kept and hovered files are used when available.

I hope this aligns with your vision for the plugin. If you have a better idea for the config option name i would be happy to change.

dedukun commented:
Hey,

Thanks for the new feature!

The way you did it is good, I just think that simply `file_pick_mode` is probably a better name, can you just change that?

nralbrecht commented:
I changed the option name to `file_pick_mode`.

While using the plugin with my change I noticed that the behavior when jumping to a bookmark that stored the parent directory was not what I expected. It only focused the directory in its parent.
I added a new `is_parent` field to each new bookmark that is used to decide wich method of jumping to the bookmark needs to be used.

If a hovered file is stored, the `is_parent` field is set to `false` and when jumping to it the `reveal` command is used to focus it.
When a parent directory is stored, the `is_parent` field is `true` and the `cd` command is used to jump to it without focusing a specific file.

dedukun commented:
Thanks for the update. 

I'll use your branch for a bit just to make sure that there's no other regression, and merge it tomorrow.

dedukun commented:
Hey,

I think that it's good the way it is, so I'll merge it.

Thanks again for the new feature.
